systemvm/debian/opt/cloud/bin/cs/CsApp.py

```diff
@@ -61,12 +61,12 @@ def setup(self):
 
         self.fw.append([
             "", "front",
-            "-A INPUT -i %s -d %s/32 -p tcp -m tcp -m state --state NEW --dport 80 -j ACCEPT" % (self.dev, self.ip)
+            "-A INPUT -i %s -s %s/32 -p tcp -m tcp -m state --state NEW --dport 80 -j ACCEPT" % (self.dev, self.ip)
```

Comment:
@Spaceman1984 should the source be the guest network ?

refer to https://github.com/apache/cloudstack/pull/3907/files

Reply:
I'm not sure I understand what you are asking. The way it is now, port 80 and 443 are only open from within the guest network.

Reply:
@Spaceman1984 
here is an example,
if VR ip is 192.168.10.1, and guest network is 192.168.10.0/24, the rule should be  `-i eth0 -s 192.168.10.0/24`, not `-i eth0 -s 192.168.10.1/32`
otherwise, the vms in guest network could not access metadata/userdata in VR.